onyo/lib/commands.py

```diff
@@ -897,63 +907,75 @@ def onyo_new(inventory: Inventory,
         If information is invalid, missing, or contradictory.
     """
 
-    from copy import deepcopy
+    from itertools import zip_longest
+    from onyo.lib.filters import Filter
+    from onyo.lib.utils import DotNotationWrapper, yaml_to_dict_multi
 
-    if auto_message is None:
-        auto_message = inventory.repo.auto_message
-
-    keys = keys or []
-    if not any([keys, edit, template, clone]):
+    if not any([keys, edit, template, clone, yaml]):
         raise ValueError("Key-value pairs or a template/clone-target must be given.")
     if template and clone:
         raise ValueError("'template' and 'clone' options are mutually exclusive.")
 
+    auto_message = auto_message or inventory.repo.auto_message
```

Comment:
Huh. This is new to me.

Just to confirm, this is a shorthand for:
```python
auto_message = inventory.repo.auto_message if auto_message is None else auto_message
```
(or more literally)
```python
auto_message = auto_message if auto_message is not None else iventory.repo.auto_message
```

Reply:
Not exactly.

It's short for `auto_message if not auto_message else inventory.repo.auto_message`. Which isn't actually what we want, so this was wrong ;-)